rasterio/plot.py

```diff
@@ -152,28 +200,25 @@ def show_hist(source, bins=10, masked=True, title='Histogram'):
     else:
         labels = (str(i + 1) for i in range(len(arr)))
 
-    # This loop should add a single plot each band in the input array,
-    # regardless of if the number of bands exceeds the number of colors.
-    # The colors slicing ensures that the number of iterations always
-    # matches the number of bands.
-    # The goal is to provide a curated set of colors for working with
-    # smaller datasets and let matplotlib define additional colors when
-    # working with larger datasets.
-    for bnd, color, label in zip_longest(arr, colors[:len(arr)], labels):
-
-        plt.hist(
-            bnd.flatten(),
-            bins=bins,
-            alpha=0.5,
-            color=color,
-            label=label,
-            range=rng
-        )
-
-    plt.legend(loc="upper right")
-    plt.title(title, fontweight='bold')
-    plt.grid(True)
-    plt.xlabel('DN')
-    plt.ylabel('Frequency')
-    fig = plt.gcf()
-    fig.show()
+    if ax:
```

Comment:
@ColinTalbert looks like duplicate logic.  Should this be

```
show = False
if not ax:
    ax = plt.gca()
```

instead?  Or should one of the `show` be True?